merino-cache/src/memory.rs

```diff
@@ -166,6 +176,11 @@ impl Suggester {
             ?removed_storage,
             "finished removing expired entries from cache"
         );
+
+        // TODO: should this also account for "pointers"?
```

Comment:
@mythmon how do we need to count the "number" of items in the deduped hashmap? Do we want to capture just the number of "stored items" like done below?